sstables/sstables.cc

```diff
@@ -1410,6 +1410,61 @@ void sstable::write_filter() {
     write_simple<component_type::Filter>(filter_ref);
 }
 
+void sstable::maybe_rebuild_filter_from_index(uint64_t num_partitions) {
+    if (!has_component(component_type::Filter)) {
+        return;
+    }
+
+    // Skip rebuilding the bloom filter if the current bitset size is between 75% and 125% of the optimal size.
```

Comment:
Using a bitset size is problematic: e.g. in some cases 25% less bits count will derive x2 increase in false positive.

And the later is what really matters. IMO we should apply the threshold (+-25%) to a configured false-positive value.

Reply:
You are right. Basing this interval on the bitset size did leave out bad filters with increased false positive rates. I have updated the code now to base this interval on the false positive rates as you have suggested.